backend/features/users/services.py

```
@@ -1,33 +1,223 @@
 from core.values.defaults import DefaultUserSettings
-from shared.utils.time_defaults import get_valid_or_default_timezone
+from shared.logger import get_logger
 from sqlalchemy.orm import Session
 
 
+class UserNotFoundError(LookupError):
+    """Raised when a user cannot be found."""
+
+
+class UsernameExistsError(ValueError):
+    """Raised when attempting to create/update to a duplicate username."""
+
+log = get_logger(__name__)
+
+
 def create_user(db: Session, username: str) -> User:
     """Create a new user account."""
     user = User(username=username)
-    # TODO: Error handling (e.g., user already exists).
-    # validate_new_user(db, user)
     db.add(user)
     db.flush()
+    log.info("Created user '%s' (id=%s)", user.username, user.id)
     return user
 
 
-def get_user_timezone(user: User) -> zoneinfo.ZoneInfo:
-    """Return the user's timezone, fall back to the default if unset."""
-    timezone_name = (
-        user.settings.timezone
-        if user.settings and user.settings.timezone
-        else DefaultUserSettings.timezone
+def create_user_with_validation(db: Session, username: str) -> User:
+    """
+    Create a new user account with validation.
+
+    Args:
+        db: Database session
+        username: Username for the new user
+
+    Returns:
+        The created user
+
+    Raises:
+        UsernameExistsError: If username already exists
+    """
+    existing_user = get_user_by_username(db, username)
+    if existing_user:
+        raise UsernameExistsError(f"Username '{username}' already exists")
+
+    return create_user(db, username)
+
+
+def get_user_by_id(db: Session, user_id: int) -> User | None:
+    """Retrieve a user by their unique ID."""
+    return db.query(User).filter(User.id == user_id).first()
+
+
+def get_user_by_username(db: Session, username: str) -> User | None:
+    """Retrieve a user by their username."""
+    return db.query(User).filter(User.username == username).first()
```

These should raise UserNotFound instead of returning None